src/site-server.ts

```diff
@@ -335,45 +339,123 @@ export class SiteServer {
 		} = {}
 	): Promise< WpCliResult > {
 		const projectPath = this.details.path;
-		const phpVersion = targetPhpVersion ?? this.details.phpVersion;
 
-		if ( ! this.wpCliExecutor ) {
-			this.wpCliExecutor = new WpCliProcess( projectPath );
-			await this.wpCliExecutor.init();
+		// If args is a string, parse it with shell-quote. If it's an array, use directly.
+		let wpCliArgs: string[];
+		if ( typeof args === 'string' ) {
+			const parsedArgs = parse( args );
+
+			// The parsing of arguments can include shell operators like `>` or `||` that the app don't support.
+			const isValidCommand = parsedArgs.every(
+				( arg: unknown ) => typeof arg === 'string' || arg instanceof String
+			);
+			if ( ! isValidCommand ) {
+				return Promise.resolve( {
+					stdout: '',
+					stderr: `Cannot execute wp-cli command with arguments: ${ args }`,
+					exitCode: 1,
+				} );
+			}
+			wpCliArgs = parsedArgs as string[];
+		} else {
+			wpCliArgs = args;
 		}
 
-		const wpCliArgs = parse( args );
+		const cliArgs: string[] = [ 'wp', '--path', projectPath ];
 
-		if ( skipPluginsAndThemes ) {
-			wpCliArgs.push( '--skip-plugins' );
-			wpCliArgs.push( '--skip-themes' );
+		if ( targetPhpVersion ) {
+			cliArgs.push( '--php-version', targetPhpVersion );
 		}
 
-		// The parsing of arguments can include shell operators like `>` or `||` that the app don't support.
-		const isValidCommand = wpCliArgs.every(
-			( arg: unknown ) => typeof arg === 'string' || arg instanceof String
-		);
-		if ( ! isValidCommand ) {
-			throw Error( `Cannot execute wp-cli command with arguments: ${ args }` );
+		cliArgs.push( ...wpCliArgs );
+
+		if ( skipPluginsAndThemes ) {
+			cliArgs.push( '--skip-plugins', '--skip-themes' );
 		}
 
-		try {
-			return await this.wpCliExecutor.execute( wpCliArgs as string[], { phpVersion } );
-		} catch ( error ) {
-			if ( ( error as MessageCanceled )?.canceled ) {
-				return {
+		const isImportExport =
+			wpCliArgs[ 0 ] === 'sqlite' && [ 'import', 'export' ].includes( wpCliArgs[ 1 ] as string );
+		const timeout = isImportExport
+			? WP_CLI_IMPORT_EXPORT_RESPONSE_TIMEOUT
+			: WP_CLI_DEFAULT_RESPONSE_TIMEOUT;
+
+		return new Promise( ( resolve ) => {
+			const result: WpCliResult = { stdout: '', stderr: '', exitCode: 1 };
+			const [ emitter ] = executeCliCommand( cliArgs, { silent: true } );
+
+			const timeoutId = setTimeout( () => {
+				resolve( {
```

Comment:
```suggestion
			const [ emitter, process ] = executeCliCommand( cliArgs, { silent: true } );

			const timeoutId = setTimeout( () => {
				process.kill( 'SIGKILL' );
				resolve( {
```
Seems like a good idea to also kill the process if the operation times out.